web/src/main/java/org/fao/geonet/monitor/link/LinkInfo.java

```diff
@@ -0,0 +1,91 @@
+package org.fao.geonet.monitor.link;
+
+import org.jdom.Element;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class LinkInfo {
+
+    private List<CheckInfo> checkInfoList;
+
+    private final Element onlineResource;
+
+    private final LinkCheckerInterface linkChecker;
+
+    public LinkInfo(Element onlineResource, LinkCheckerInterface linkChecker) {
+        checkInfoList = Collections.synchronizedList(new ArrayList<CheckInfo>());
+        this.onlineResource = onlineResource;
+        this.linkChecker = linkChecker;
+    }
+
+    public int getCheckCount() {
+        return checkInfoList.size();
+    }
+
+    public LinkMonitorService.Status getStatus() {
+        if (getCheckCount() < LinkMonitorService.maxChecks) {
+            if (getCheckCount() > 0 && ! hasSuccessCheck()) {
+                // In the case when we don't have enough checks, but have more
+                // than one check and they all failed - mark link as failed
+                return LinkMonitorService.Status.FAILED;
+            } else {
+                // Not enough information to decide whether the link is online
+                return LinkMonitorService.Status.UNKNOWN;
+            }
+        }
+
+        long now = System.currentTimeMillis() / 1000l;
+        long lastCheck = checkInfoList.get(getCheckCount() - 1).timestamp;
+        if (now - lastCheck > LinkMonitorService.freshness) {
+            LinkMonitorService.getLogger().debug(String.format("Last test is too old, ('%s' seconds ago)", now - lastCheck));
+            return LinkMonitorService.Status.UNKNOWN;
+        }
+
+        int successChecks = 0;
+        for (final CheckInfo checkInfo : checkInfoList) {
+            if (checkInfo.status) {
+                successChecks++;
+            }
+        }
+
+        int percentSuccess = 100 * successChecks / getCheckCount();
+
+        if (percentSuccess >= LinkMonitorService.percentWorking) {
```

Comment:
`percentWorking` may be better named `percentWorkingThreshold` (if that's indeed what it is)